types/process.go

```diff
@@ -44,6 +44,10 @@ type FileDescriptor interface {
 	FileDescriptorCount() (int, error)
 }
 
+type OpenHandles interface {
+	OpenHandlesCount() (int, error)
```

Comment:
Does this differ enough from `FileDescriptor` to justify it being its own type? To me knowing the FD count is the same thing as knowing the Handle count, but perhaps I don't understand the differences.

If not implementing the `FileDescriptors()` part of `FileDescriptor` is the issue then you could decompose the interface more.

Reply:
I introduced a new interface, so if in the future we want to add something platform specific, we don't need to split it. Although file descriptor and file handles are nearly the same, I would not use the same or the current interface. What do you think about another name which is not tied to any platform? For example `ProcessOpenFiles`? Or separating the interface into `OpenFileCounter` and `OpenFileEnumerator`? Or `OpenFileCounter` and `OpenFilesLister`?

Reply:
> What do you think about another name which is not tied to any platform?

Yes, I'd like this. I think "handle" is a more generic name than "file descriptor" given that it encompasses more things than just files.

> Although file descriptor and file handles are nearly the same, I would not use the same or the current interface.

I agree about not using the current interface, but as a user I would want to use the same interface (or interfaces) to get this data from all platforms.

> Or separating the interface into OpenFileCounter and OpenFileEnumerator?

That's pretty close to what I'm thinking, but with the term "Handle" instead of "File" because it's not just files being counted.

```
type OpenHandleCounter interface {
    OpenHandleCount() (int, error)
}

type OpenHandleEnumerator struct {
    OpenHandles() ([]string, error)
}
```